R/list_animal_ids_api.R

```diff
@@ -0,0 +1,27 @@
+#' List all available animal ids
+#'
+#' @param credentials Login credentials to the ETN database, as created by
+#'   `get_credentials()`
+#'
+#' @return A vector of all unique `id_pk` present in `common.animal_release`.
+#'
+#' @export
+list_animal_ids_api <- function(credentials = get_credentials()) {
+  api_domain <- "https://opencpu.lifewatch.be"
+  function_path <- "library/etn/R/list_animal_ids"
+
+  # POST the function request to the api_domain, request json as direct return
+  response <-
+    httr::POST(stringr::str_glue(api_domain, function_path, "json", .sep = "/"),
+      body = list(
+        con = credentials
+      )
+    )
+  # If request was not succesful, generate a warning
+  httr::warn_for_status(response, "submit request to API server")
```

Comment:
When will this warn?

- `https://opencpu.lifewatch.com/library/etn/R/list_animal_ids/json` (.com) returns error when calling
- `https://opencpu.lifewatch.be/library/etn/R/list_animal_ids/js` (js) doesn't warn

Reply:
It'll warn for 40x http response codes, since you passed an invalid content type as a post request (`js`), opencpu falls back to the POST -> GET protocol (instead of returning json) you get a string array, thus content-type text instead of application/json, but no warning since the call was succesful (HTTP 201)

We could add an extra test that checks if the server returns the right content type.

Reply:
I've added an extra test for this, but again, this should probably be wrapped in a helper function in the future as we want to check this for every request.

Reply:
Yes, good to do this for the helper function.

Reply:
Done